dbt/include/snowflake/macros/adapters.sql

```diff
@@ -72,14 +72,16 @@
 )
 {% endmacro %}
 
-{% macro snowflake__create_view_as(relation, sql) -%}
+{% macro snowflake__create_view_as_with_temp_flag(temporary, relation, sql) -%}
```

Comment:
Part 1 of 3 of multi-part suggestion (completely untested!).

Basically, make the `temporary` field the final parameter and give it a default value.

```suggestion
{% macro snowflake__create_view_as_with_temp_flag(relation, sql, temporary=True) -%}
```

If this suggestion works, you could consider just using this as the new definition of `snowflake__create_view_as` and skip the independent `snowflake__create_view_as_with_temp_flag` macro altogether.